README.md

```diff
@@ -112,9 +112,25 @@ org.freedesktop.tuhi1.Device
   Property: Listening (b)
       Indicates whether the daemon is currently listening for the device.
 
-      This property is set to True when a Listen() request initiates the
-      search for device connections. When the Listen() request completes
-      upon timeout, the property is set to False.
+      This property is set to True when a StartListening() request initiates
+      the search for device connections. When the StartListening() request
+      completes upon timeout, or when StopListening() is called, the property
+      is set to False.
+
+      When the user press the button on the device, the daemon shall connect
+      to the device and the daemon downloads all drawings from the
+      device. The daemon then disconnects from the device.
+
+      If successfull, the drawings are deleted from the device. The data is
+      held by the daemon in non-persistent storage until the daemon is stopped
+      or we run out of memory, whichever happens earlier.
+      Use GetJSONData() to retrieve the data from the daemon.
+
+      When drawings become available from the device, the DrawingsAvailable
+      property updates to the number of available drawings.
+      When the button is pressed multiple times, any new data is appended
+      to the existing list of drawings as long as this property is True.
+
```

Comment:
We have a `Listening` property but also a `StoppedListening` signal - do we need both? It seems superfluous and while there's plenty of debugging use-cases where having the property is useful, the primary use-case for this is still one client doing everything, so the signal is enough. And the signal is what can carry the error code, so we do need that.

Reply:
IIRC, I wanted to keep `Listening` for the use case where you have 2 clients, and it's a way of notifying the other client that the first one started listening on the device.

OTOH, the `StoppedListening` signal is meant for the client that requested the `StartListening`.
However, we should also target the `StoppedListening` signal to the client, because if one client has a grab on the device (see #11), its signal should be different than the other clients who tried to steal the grab.

Reply:
Not sure how much effort it would be but we could still internally separate the various timeouts so you get the `ListeningStopped` signal even while the device itself is still listening and handling events for the other client.

Reply:
Quick answer while trying to focus on sth else: the timeouts are not required anymore. In this branch we are tracking clients, so if a client disappears, we call ~StopListening()` for it and this makes the whole timeouts moot